src/components/ArticleById.jsx

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from "react";
+import {getArticleId, getUsers, patchUpvote, patchDownvote} from '../utils/api'
+import {useParams} from 'react-router-dom'
+import {Link} from 'react-router-dom'
+import Comments from '../components/Comments'
+const ArticleById = () => {
+const [article, setArticle] = useState([])
+const [user, setUser] = useState([])
+const [votes, setVotes] = useState(0)
+const {article_id} = useParams()
+
+
+useEffect(() => {
+    getArticleId(article_id).then((article) => {
+        setArticle(article.article)
+    })
+},[article_id])
+useEffect(() => {
+    getUsers().then((user) => { 
+        setUser(user.users)
+    })
+},[])
+
+
+const likeButton = (e) => {
+e.preventDefault();
+setVotes((currVotes) => currVotes + 1)
+patchUpvote(article_id)
+}
+const disLikeButton = (e) => {
+    e.preventDefault();
+    setVotes((currVotes) => currVotes - 1)
+    patchDownvote(article_id)
+    }
+
+let image; 
+    for (let i = 0; i<user.length; i++) {
+      if(article.author === user[i].username) {
+            image = user[i].avatar_url
+      }
+  }
```

Comment:
Don't loop through all users just to get their image. Make a separate API call for the specific user to access their avatar_url instead.